source/extensions/common/wasm/wasm.cc

```diff
@@ -2841,6 +2840,12 @@ std::shared_ptr<Wasm> getOrCreateThreadLocalWasm(Wasm& base_wasm, absl::string_v
   auto wasm = getThreadLocalWasmPtr(base_wasm.vm_id());
   if (wasm) {
     auto root_context = wasm->start();
+    // NB: we are reusing this VM with the new configuration so we need to replace the plugin which
+    // contains a Scope& which is in a ListenerImpl which may be deleted. The Scope has no prefix
+    // (""), and does nothing but provide a layer of caching and forward to the underlying
+    // Stats::ThreadLocalStoreImpl.
+    // TODO(jplevyak): this will go away after https://github.com/envoyproxy/envoy-wasm/issues/258
```

Comment:
What does base_wasm.plugin() return? Does this only affect wasm or also null sandbox?

Can you add a bit of explanation? I see this one extra line and a change of name?

Reply:
base_wasm.plugin() returns a plugin with a valid Scope.  As commented, this is going to change when we move the VM declarations out of the the plugin configuration.  At that point we will have control of the lifetime and we will not have a plugin which is generic to the Wasm and instead we will only have the plugin in the Context (as it should be).